legal-api/tests/unit/services/filings/validations/change_of_director/test_validation_effective_date.py

```diff
@@ -56,130 +50,170 @@ def test_effective_date_sanity_check(session):
+    yesterday = datetime(2001, 8, 4, 12, 0, 0, 0, tzinfo=timezone.utc)
+    effective_date = as_effective_date(yesterday)
+    filing['filing']['header']['effectiveDate'] = effective_date.isoformat()
     with freeze_time(now):
         err = validate_effective_date(business, filing)
     assert not err
 
 
 def test_validate_effective_date_not_before_founding(session):
-    """Assert the filing is not before the business was founded.
-
-    Rules:
-        - The effective date cannot be a date prior to their Incorporation Date
-    """
+    """Assert that the effective date cannot be a date prior to their Incorporation Date."""
     # setup
     identifier = 'CP1234567'
-    now = datetime(2001, 8, 5, 0, 0, 0, 0, tzinfo=timezone.utc)
-
+    now = datetime(2001, 8, 5, 12, 0, 0, 0, tzinfo=timezone.utc)
     business, filing = common_setup(identifier, now)
 
-    # The effective date cannot be a date prior to their Incorporation Date
-    effective_date = now - datedelta.DAY
-    business.founding_date = now
+    # The effective date _cannot_ be before their Incorporation Date.
+    before = datetime(2000, 8, 4, 12, 0, 0, 0, tzinfo=timezone.utc)
+    effective_date = as_effective_date(before)
     filing['filing']['header']['effectiveDate'] = effective_date.isoformat()
-
     with freeze_time(now):
         err = validate_effective_date(business, filing)
     assert err
 
-    business.founding_date = now - datedelta.YEAR
+    # The effective date _can_ be on their Incorporation Date.
+    on = datetime(2000, 8, 5, 12, 0, 0, 0, tzinfo=timezone.utc)
+    effective_date = as_effective_date(on)
+    filing['filing']['header']['effectiveDate'] = effective_date.isoformat()
+    with freeze_time(now):
+        err = validate_effective_date(business, filing)
+    assert not err
+
+    # The effective date _can_ be after their Incorporation Date.
+    after = datetime(2000, 8, 6, 12, 0, 0, 0, tzinfo=timezone.utc)
+    effective_date = as_effective_date(after)
+    filing['filing']['header']['effectiveDate'] = effective_date.isoformat()
     with freeze_time(now):
         err = validate_effective_date(business, filing)
     assert not err
 
 
 def test_validate_effective_date_not_before_other_COD(session):  # noqa: N802; COD is an acronym
-    """Assert that the new filing is not before an existing COD.
-
-    Rules:
-       - The effective date of change cannot be a date that
-            is farther in the past as a previous COD filing(Standalone or AR).
-    """
+    """Assert that the effective date of change cannot be before a previous COD filing."""
     # setup
     identifier = 'CP1234567'
-    now = datetime(2001, 8, 5, 0, 0, 0, 0, tzinfo=timezone.utc)
-    filing_cod = copy.deepcopy(FILING_HEADER)
-    filing_cod['filing']['header']['name'] = 'changeOfDirectors'
-    filing_cod['filing']['changeOfDirectors'] = copy.deepcopy(CHANGE_OF_DIRECTORS)
-
+    founding_date = datetime(2000, 8, 5, 12, 0, 0, 0, tzinfo=timezone.utc)
```

Comment:
Why all the setup change? Styling preference?

Reply:
Not at all. I am being more explicit (clearer) about initial conditions and what each test verifies. I also renamed some variables to reduce confusion (eg, "now").
(I figure code that's easy to understand/review has a better chance of being correct ;) )

Reply:
I guess that's a matter of opinion. more compact is clearer to lots of folks too.
As before, generally our accepted rule is everythings open in refactor, reformatting for preference is something we're avoiding.
It's also MUCH harder to see what you actually changed wrt to business or functionality, as opposed to just moved around.